src/modules/account/RadixTokenDefinitionAccountSystem.ts

```diff
@@ -49,15 +50,49 @@ export class RadixTokenDefinitionAccountSystem implements RadixAccountSystem {
         }
         this.processedAtomHIDs.set(atom.hid.toString(), true)
 
-        for (const spunParticle of atom.getParticles()) {
-            if (spunParticle.particle instanceof RadixTokenDefinitionParticle && spunParticle.spin === RadixSpin.UP) {
-                this.createOrUpdateTokenDefinition(spunParticle.particle)
-            } else if (spunParticle.particle instanceof RadixMintedTokensParticle && spunParticle.spin === RadixSpin.UP) {
-                const particle = (spunParticle.particle as RadixMintedTokensParticle)
-                this.updateTotalSupply(particle.getTokenDefinitionReference(), particle.getAmount())
-            } else if (spunParticle.particle instanceof RadixBurnedTokensParticle && spunParticle.spin === RadixSpin.UP) {
-                const particle = (spunParticle.particle as RadixBurnedTokensParticle)
-                this.updateTotalSupply(particle.getTokenDefinitionReference(), particle.getAmount().neg())
+
+        for (const particleGroup  of atom.getParticleGroups()) {
+            let tokenDefinition: RadixTokenDefinition
+
+            if (particleGroup.containsParticle(RadixTokenDefinitionParticle)) {
+                // Token definition
+                for (const spunParticle of particleGroup.getParticles()) {
+                    if (spunParticle.particle instanceof RadixTokenDefinitionParticle && spunParticle.spin === RadixSpin.UP) {
+                        this.createOrUpdateTokenDefinition(spunParticle.particle)
+                    } else if (spunParticle.particle instanceof RadixUnallocatedTokensParticle) {
+                        const particle = (spunParticle.particle as RadixUnallocatedTokensParticle)
+                        tokenDefinition = this.getOrCreateTokenDefinition(particle.getTokenDefinitionReference())
+
+                        if (spunParticle.spin === RadixSpin.UP) {
+                            tokenDefinition.unallocatedTokens.set(particle.getHID().toString(), particle)
+                        } else {
+                            tokenDefinition.unallocatedTokens.delete(particle.getHID().toString())
+                        }
+                    } 
+                }
+
+            } else if (
+                particleGroup.containsParticle(RadixUnallocatedTokensParticle)
+                && particleGroup.containsParticle(RadixTransferrableTokensParticle)) {
```

Comment:
In this case yes, because that's how I can identify that this is a mint or burn action, the token definition action for example has a TokenDefinitionParticle and an UnallocatedTokensParticle, but no Transferrable particle. In practice that case is currently excluded with the else if, but I thought I'd rather be explicit about the requirements.